document/runproperties.go

```diff
@@ -244,3 +244,187 @@ func (r RunProperties) SetVerticalAlignment(v sharedTypes.ST_VerticalAlignRun) {
+		} else if fonts.CsAttr != nil {
+			return *fonts.CsAttr
+		}
+	}
+	return ""
+}
+
+// EastAsiaFont returns the name of run font family for East Asia.
+func (r RunProperties) EastAsiaFont() string {
+	if fonts := r.x.RFonts; fonts != nil {
+		if fonts.EastAsiaAttr != nil {
+			return *fonts.EastAsiaAttr
+		}
+	}
+	return ""
+}
+
+// ColorStr returns the hex value of run color.
+func (r RunProperties) ColorStr() string {
+	if color := r.x.Color; color != nil {
+		valAttr := color.ValAttr
+		if valAttr.ST_HexColorRGB != nil {
+			return *valAttr.ST_HexColorRGB
+		}
+	}
+	return ""
+}
+
+// SpacingValue returns the value of run spacing.
+func (r RunProperties) SpacingValue() int64 {
+	if spacing := r.x.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.Int64 != nil {
+			return *valAttr.Int64
+		}
+	}
+	return int64(0)
+}
+
+// SpacingMeasure returns the measure of run spacing.
+func (r RunProperties) SpacingMeasure() string {
+	if spacing := r.x.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.ST_UniversalMeasure != nil {
+			return *valAttr.ST_UniversalMeasure
+		}
+	}
+	return ""
+}
+
+// SizeValue returns the value of run font size.
+func (r RunProperties) SizeValue() float64 {
+	if sz := r.x.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// SizeMeasure returns the measure of run font size.
+func (r RunProperties) SizeMeasure() string {
+	if sz := r.x.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// ComplexSizeValue returns the value of run font size for complex fonts.
+func (r RunProperties) ComplexSizeValue() float64 {
+	if szCs := r.x.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// ComplexSizeMeasure returns the measure of run font size for complex fonts.
+func (r RunProperties) ComplexSizeMeasure() string {
+	if szCs := r.x.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// UnderlineType returns the type of run underline.
+func (r RunProperties) UnderlineType() wml.ST_Underline {
+	if underline := r.x.U; underline != nil {
+		return underline.ValAttr
+	}
+	return 0
+}
+
+// UnderlineColor returns the hex color value of run underline.
+func (r RunProperties) UnderlineColor() string {
+	if underline := r.x.U; underline != nil {
+		color := underline.ColorAttr
+		if color.ST_HexColorRGB != nil {
+			return *color.ST_HexColorRGB
+		}
+	}
+	return ""
+}
+
+// VertAlign returns the value of run vertical align.
+func (r RunProperties) VertAlign() sharedTypes.ST_VerticalAlignRun {
```

Comment:
Would it make sense to encapsulate by higher level type?  Any similar cases in the codebase?